### PR TITLE
Award Nomination App. Cancel, Add watchers, Add Attachments

### PR DESCRIPTION
**Pull request by alenaminsk**
Award Nomination App. Cancel, Add watchers, Add Attachments

### CODE REVIEW

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java
Comment (katm007) [inline src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:1]: On line 36, why was the spelling changed?
Please remove the extra spaces in this file. Are the commented out lines 30-32 needed?

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java
Comment (katm007) [inline src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java:1]: Why are extra letters added to these steps?

File: src/test/java/PLATFORM_BUSINESS/RCMDB/steps/BusinessCaseFormSteps.java
Comment (katm007) [inline src/test/java/PLATFORM_BUSINESS/RCMDB/steps/BusinessCaseFormSteps.java:1]: Same here, why was the spelling changed? Can lines 275-285 be removed if they're no longer needed?

File: src/test/java/ServiceNow/AwardNomination/StepsImplementation/AwardNominationStepsImplementation.java
Comment (katm007) [inline src/test/java/ServiceNow/AwardNomination/StepsImplementation/AwardNominationStepsImplementation.java:1]: Please remove the extra spaces on lines 14 and 16

**Review by lfcabrera14 — COMMENTED** [review]
please address comments

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:1]: updated

Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:1]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public void clickonCADSRRequest() throws InterruptedException {
 	@Then("User should see default columns for incident grid")
 	public void viewDefaultColumns() throws InterruptedException {
 		nativeViewPage.viewDefaultColumns();
-		
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:52]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 2]
```diff
@@ -118,32 +118,32 @@ public void elevatePermission() throws InterruptedException {
 	@Then("User should see Save, Add Attachments and Submit button")
 	public void user_should_see_Save_Add_Attachments_and_Submit_button() {
 		nativeViewPage.switchToFrame();
-		
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:121]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 3]
```diff
@@ -118,32 +118,32 @@ public void elevatePermission() throws InterruptedException {
...
 
 	}
 
 	@Then("Start a new request on nativ view")
 	public void start_new_Requst() {
 		nativeViewPage.startANewRequest();
 	}
-	
-	
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:136]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 4]
```diff
@@ -118,32 +118,32 @@ public void elevatePermission() throws InterruptedException {
...
+
 	@Then("User should see available different choices of status")
 	public void validateStatusCHoices() throws Exception {
 		nativeViewPage.allSelectOptions();
-		
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:141]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 5]
```diff
@@ -155,20 +155,20 @@ public void set_assigned_to() {
 
 	@Then("Validate assigned to value in child ticket")
 	public void validate_assigned_to() throws InterruptedException {
-		
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:158]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 6]
```diff
@@ -155,20 +155,20 @@ public void set_assigned_to() {
...
 		String value = nativeViewPage.getAssignedToValue();
 		Assert.assertTrue("Assigned to value in child ticket is not correct", "Janice Knable".equalsIgnoreCase(value));
 	}
 
 	@Then("User should see Status field")
 	public void user_should_see_Status_field() {
 		nativeViewPage.validateTheLabelAsStatus();
-		
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:166]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 7]
```diff
@@ -155,20 +155,20 @@ public void set_assigned_to() {
...
 	}
 
 	@Then("Validate parent ticket reference")
 	public void validatePerenetTicketReference() {
-		
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:171]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 8]
```diff
@@ -179,7 +179,7 @@ public void validateTextOnLandingPage() {
 
 	@Then("validate text on request form page")
 	public void validateTextonRequestFormPage() {
-		
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:182]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 9]
```diff
@@ -191,12 +191,12 @@ public void enterOneMoreEmailId() {
 	@Then("validate page footer")
 	public void validatePageFooter() {
 		nativeViewPage.checkIfPageFooterISasExpected();
-		
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:194]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java [hunk 10]
```diff
@@ -191,12 +191,12 @@ public void enterOneMoreEmailId() {
...
 	}
 
 	@Then("validate that the list of organiation is as expected")
 	public void validateTheListOfOrganization() throws Exception {
 		nativeViewPage.validateTheListOfOrganization();
-		
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/NativeViewSteps.java:200]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java:1]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java [hunk 11]
```diff
@@ -449,8 +449,8 @@ public void assertCurationCountAs10() {
 
 	}
 
-	@Then("Verify Email notification with subject as {string}")
-	public void verifyEmail(String subject) throws InterruptedException {
+	@Then("Verify Email notificationkkk with subject as {string}")
+	public void verifyEmailkkk(String subject) throws InterruptedException {
 		boolean status = servicePortalPage.verifyEmail(subject);
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java:455]: updated

File: src/test/java/PLATFORM_BUSINESS/RCMDB/steps/BusinessCaseFormSteps.java
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/RCMDB/steps/BusinessCaseFormSteps.java:1]: updated

File: src/test/java/PLATFORM_BUSINESS/RCMDB/steps/BusinessCaseFormSteps.java [hunk 12]
```diff
@@ -272,15 +272,15 @@ public void Fill_all_the_details() throws InterruptedException {
...
-		if (loginStepsImpl.isLoginButtonDisplayed()) {
-			loginStepsImpl.clckOnLoginButton();
-		}
-		loginStepsImpl.enterUsername("");
-		String decyptedPass = EncryptionUtils.decrypt("");
-		loginStepsImpl.enterPassword(decyptedPass);
-		CucumberLogUtils.logScreenshot();
-		loginStepsImpl.clickOnSignInButton();
-	}
+//	@When("User logins to the application for RCMDB")
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/RCMDB/steps/BusinessCaseFormSteps.java:275]: updated

File: src/test/java/PLATFORM_BUSINESS/RCMDB/steps/PortalViewSteps.java [hunk 13]
```diff
@@ -20,12 +20,12 @@ public void before() {
...
+
 	@Given("User opens IT Procurement Governance Application in browser")
 	public void openATOApp() {
 		portalViewPage.openApp();
 	}
-	
+
 	@When("User Logins to the application")
 	public void login() throws Exception {
 		if (loginStepsImpl.isLoginButtonDisplayed()) {
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/RCMDB/steps/PortalViewSteps.java:31]: updated

File: src/test/java/ServiceNow/AwardNomination/StepsImplementation/AwardNominationStepsImplementation.java
Comment (alenaminsk) [reply src/test/java/ServiceNow/AwardNomination/StepsImplementation/AwardNominationStepsImplementation.java:1]: updated

File: src/test/java/ServiceNow/AwardNomination/StepsImplementation/AwardNominationStepsImplementation.java [hunk 14]
```diff
@@ -113,4 +113,72 @@ public void clickSubmitButton() {
...
+
+    /**
+     * verify Watcher is added
+     * @param watcher
+     */
+    public void verifyWatcherAdded(String watcher) {
+        CommonUtils.waitForVisibility(awardNominationPage.watchListField);
+        softAssert.assertTrue(awardNominationPage.watchListField.getText().equals(watcher));
+    }
+
```
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/ServiceNow/AwardNomination/StepsImplementation/AwardNominationStepsImplementation.java:178]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java [hunk 15]
```diff
@@ -360,53 +337,45 @@ public void verifyDisabledFields() throws Exception {
...
 
 	@Then("User should see default priority as Moderate")
 	public void validatePriority() throws InterruptedException {
 		servicePortalPage.validatePriority();
-
 	}
 
 	@Then("Go to native view")
-	public void getNativeView() {
+	public void getNativeViewm() {
```
Comment (katm007) [inline src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java:378]: is this change in spelling needed?

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java [hunk 16]
```diff
@@ -446,15 +415,12 @@ public void updateCurationCOunt() {
 	@Then("User should see updated curation count for parent ticket")
 	public void assertCurationCountAs10() {
 		servicePortalPage.verifyCurationCOunt();
-
 	}
 
-	@Then("Verify Email notification with subject as {string}")
-	public void verifyEmail(String subject) throws InterruptedException {
+	@Then("Verify Email notificationkkk with subject as {string}")
```
Comment (katm007) [inline src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java:420]: is this change in spelling needed?

File: src/test/java/ServiceNow/AwardNomination/Features/VIBEAppreciationRequest.feature [hunk 17]
```diff
@@ -17,5 +17,27 @@
...
+     When User clicks Add Watchers button
+     And User adds "Anna Nicholson" as a watcher
+     Then User can see that Watch List ticket field is populated with "Anna Nicholson" name
+     When User adds attachment
+     Then User can verify that attachment is added
+
+
+
+
+
```
Comment (sgugulothu) [inline src/test/java/ServiceNow/AwardNomination/Features/VIBEAppreciationRequest.feature]: Can you remove empty lines 37-41

File: src/test/java/ServiceNow/AwardNomination/Pages/AwardNominationPage.java [hunk 18]
```diff
@@ -67,6 +67,46 @@ public class AwardNominationPage {
     @FindBy(xpath = "//span[.=' Tickets ']")
     public WebElement sectionTickets;
 
+    /**  Last submitted VIBE Ticket */
+    @FindBy(xpath = "(//h4[.='VIBE Peer-to-Peer Appreciation Program'])[1]")
+    public WebElement lastSubmittedVIBETicket;
+
```
Comment (sgugulothu) [inline src/test/java/ServiceNow/AwardNomination/Pages/AwardNominationPage.java:73]: Can you remove empty line 73

File: src/test/java/ServiceNow/AwardNomination/Pages/AwardNominationPage.java [hunk 19]
```diff
@@ -67,6 +67,46 @@ public class AwardNominationPage {
...
+    public WebElement addAttachmentButton;
+
+    /**  Cancel My Ticket modal */
+    @FindBy(xpath = "//h3[.='Cancel My Ticket']")
+    public WebElement modalCancelMyTicket;
+
+    /**  text Field Add Watchers */
+    @FindBy(xpath = "//input[@class='select2-input select2-default']")
+    public WebElement textFieldAddWatchers;
+
```
Comment (sgugulothu) [inline src/test/java/ServiceNow/AwardNomination/Pages/AwardNominationPage.java:101]: Can you remove empty line 101

File: src/test/java/ServiceNow/AwardNomination/Pages/AwardNominationPage.java [hunk 20]
```diff
@@ -67,6 +67,46 @@ public class AwardNominationPage {
...
+    /**  Save Changes confirmation modal button */
+    @FindBy(xpath = "//span[contains(text(),'Save Changes')]")
+    public WebElement saveChangesConfModalButton;
+
```
Comment (sgugulothu) [inline src/test/java/ServiceNow/AwardNomination/Pages/AwardNominationPage.java:105]: Can you remove empty line 105

File: src/test/java/ServiceNow/AwardNomination/Steps/AwardNominationSteps.java [hunk 21]
```diff
@@ -1,10 +1,13 @@
 package ServiceNow.AwardNomination.Steps;
 
 import appsCommon.PageInitializers.PageInitializer;
+import com.nci.automation.web.CommonUtils;
 import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
```
Comment (sgugulothu) [inline src/test/java/ServiceNow/AwardNomination/Steps/AwardNominationSteps.java]: remove line

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java (hunk 15 above)
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java:378]: updated

File: src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java (hunk 16 above)
Comment (alenaminsk) [reply src/test/java/PLATFORM_BUSINESS/CADSR/steps/ServicePortalSteps.java:420]: updated

**Review by sgugulothu — APPROVED** [review]

**Review by Mariachaudhry — APPROVED** [review]

**Review by lfcabrera14 — APPROVED** [review]
approved

**Pull request merged**